Nightly failing on FreeBSD

## Required information

**Operating system:** FreeBSD 15.0

**Rust version:** 1.94.0

**iceoryx2 version:** main

**Detailed log output:**

```
  error[E0425]: cannot find value `os_specific_buffer` in this scope
    --> iceoryx2-pal/posix/src/freebsd/timer.rs:23:50
     |
  23 |     crate::internal::timer_create(clock_id, &mut os_specific_buffer, timer_id)
     |                                                  ^^^^^^^^^^^^^^^^^^ not found in this scope
```


**Observed result or behaviour:**
https://github.com/eclipse-iceoryx/iceoryx2/actions/runs/23224944240/job/67505882631

**Expected result or behaviour:**
Build is passing on FreeBSD

**Conditions where it occurred / Performed steps:**
Compile iceoryx2-pal-posix on FreeBSD 15

--- a/doc/release-notes/iceoryx2-unreleased.md
+++ b/doc/release-notes/iceoryx2-unreleased.md
@@ -71,6 +71,8 @@
   [#1444](https://github.com/eclipse-iceoryx/iceoryx2/issues/1444)
 * Fix libc dependency version
   [#1447](https://github.com/eclipse-iceoryx/iceoryx2/issues/1447)
+* Fix FreeBSD build
+  [#1455](https://github.com/eclipse-iceoryx/iceoryx2/issues/1455)
 
 ### Refactoring
 

--- a/iceoryx2-pal/posix/src/freebsd/timer.rs
+++ b/iceoryx2-pal/posix/src/freebsd/timer.rs
@@ -20,7 +20,7 @@ pub unsafe fn timer_create(
     sevp: *mut sigevent,
     timer_id: *mut timer_t,
 ) -> int {
-    crate::internal::timer_create(clock_id, &mut os_specific_buffer, timer_id)
+    crate::internal::timer_create(clock_id, sevp, timer_id)
 }
 
 pub unsafe fn timer_delete(timer_id: timer_t) -> int {

